tasks/Task.cpp

```diff
@@ -16,17 +16,69 @@ Task::~Task()
 {
 }
 
+static RigidBodyState convertToRBS(Status const& data)
+{
+    RigidBodyState rbs;
+    rbs.time = data.timestamp;
+    rbs.position = Eigen::Vector3d(0, 0, data.environment.depth / 10.);
+    rbs.orientation =
+        Eigen::Quaterniond(Eigen::AngleAxisd(data.attitude.yaw / 10. / 180.0 * M_PI,
+                               Eigen::Vector3d::UnitZ()) *
+                           Eigen::AngleAxisd(data.attitude.pitch / 10. / 180.0 * M_PI,
+                               Eigen::Vector3d::UnitY()) *
+                           Eigen::AngleAxisd(data.attitude.roll / 10. / 180.0 * M_PI,
+                               Eigen::Vector3d::UnitX()));
+    return rbs;
+}
+
 static RigidBodyState convertToRBS(PingStatus const& data)
 {
     RigidBodyState rbs;
 
     rbs.time = data.timestamp;
-    rbs.position = Eigen::Vector3d(data.response.acoustic_fix.position.north,
-        -data.response.acoustic_fix.position.east,
-        -data.response.acoustic_fix.position.depth);
+    rbs.position = Eigen::Vector3d(data.response.acoustic_fix.position.north / 10.0,
+        data.response.acoustic_fix.position.east / 10.0,
+        data.response.acoustic_fix.position.depth / 10.0);
     return rbs;
 }
```

Comment:
I think naming both functions the same is confusing, if we come back to this in a couple of months we would curse our previous selves. Just name the first one to convertToZWithOrientationRBS or something like that and the second to convertToPositionRBS